docs/wcf-config.MD

```diff
@@ -0,0 +1,200 @@
+# WCF Instrumentation Configration
+
+For each version of .NET the `OpenTelemetry.Instrumentation.Wcf` package version
+has to match the version used by Autoinstrumentation library to avoid compatibility
+issues.
+
+## WCF Client Configuration (.NET Framework)
+
+For this integration you shouldn't attach `OpenTelemetry.Instrumentation.Wcf` package.
+Attaching it can result in `API` version conflict and application crash.
+Please use `ExludeAssets` for compilation purposes.
+
+```xml
+  <ItemGroup>
+    <PackageReference Include="OpenTelemetry.Instrumentation.Wcf" Version="1.0.0-rc.7" ExcludeAssets="runtime" />
+  </ItemGroup>
+```
+
+Add the `IClientMessageInspector` instrumentation via a behavior extension on
+the clients you want to instrument:
+
+```xml
+<?xml version="1.0" encoding="utf-8" ?>
+<configuration>
+  <system.serviceModel>
+    <extensions>
+      <behaviorExtensions>
+        <add name="telemetryExtension" type="OpenTelemetry.Instrumentation.Wcf.TelemetryEndpointBehaviorExtensionElement, OpenTelemetry.Instrumentation.Wcf" />
+      </behaviorExtensions>
+    </extensions>
+    <behaviors>
+      <endpointBehaviors>
+        <behavior name="telemetry">
+          <telemetryExtension />
+        </behavior>
+      </endpointBehaviors>
+    </behaviors>
+    <bindings>
+      <basicHttpBinding>
+        <binding name="basicHttpConfig">
+          <security mode="None" />
+        </binding>
+      </basicHttpBinding>
+      <netTcpBinding>
+        <binding name="netTCPConfig">
+          <security mode="None" />
+        </binding>
+      </netTcpBinding>
+    </bindings>
+    <client>
+      <endpoint address="http://localhost:9009/Telemetry" binding="basicHttpBinding" bindingConfiguration="basicHttpConfig" behaviorConfiguration="telemetry" contract="TestApplication.Wcf.Shared.IStatusServiceContract" name="StatusService_Http" />
+    </client>
+  </system.serviceModel>
+</configuration>
+```
+
+Example project available in
+[test/test-applications/integrations/TestApplication.Wcf.Client.NetFramework](../test/test-applications/integrations/TestApplication.Wcf.Client.NetFramework/)
+folder.
+
+## WCF Client Configuration (.NET Core)
+
+Add the `IClientMessageInspector` instrumentation as an endpoint behavior on the
+clients you want to instrument:
+
+```csharp
+StatusServiceClient client = new StatusServiceClient(binding, remoteAddress);
```

Comment:
Good candidate for bytecode intrumentation.

Reply:
I would actually move it to a separate PR

Reply:
Please create follow up ticket.

Reply:
https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/1143